hibernate-core/src/test/java/org/hibernate/orm/test/annotations/collectionelement/PrimitiveArrayElementCollectionTempTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.annotations.collectionelement;
+
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OrderColumn;
+
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @author Andrea Boriero
+ */
+@DomainModel(
+		annotatedClasses = {
+				PrimitiveArrayElementCollectionTempTest.Favorite.class
+		}
+)
+@SessionFactory
+public class PrimitiveArrayElementCollectionTempTest {
+
+	@BeforeEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Favorite favorite = new Favorite( 1L );
+					int[] favoriteNumbers = new int[2];
+					favoriteNumbers[0] = 3;
+					favoriteNumbers[1] = 7;
+					favorite.setFavoriteNumbers( favoriteNumbers );
+					session.persist( favorite );
+				}
+		);
+	}
+
+	@Test
+	public void testIt(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Favorite favorite = session.get( Favorite.class, 1L );
+					int[] favoriteNumbers = favorite.getFavoriteNumbers();
+					assertThat( favoriteNumbers.length, is( 2 ) );
+					assertThat( favoriteNumbers[0], is( 3 ) );
+					assertThat( favoriteNumbers[1], is( 7 ) );
```

Comment:
We might well use Hamcrest's array matching matcher. e.g. http://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/collection/ArrayMatching.html#arrayContaining-E...-